tests/phpunit/elementor/core/app/import-export/test-import.php

```diff
@@ -0,0 +1,479 @@
+		$import->register( new Elementor_Content() );
+		$import->register( new Wp_Content() );
+
+		// Act
+		$result = $import->run();
+
+		// Assert
+		$this->assertCount( 1, $result['content']['post']['succeed'] );
+		$this->assertCount( 1, $result['content']['page']['succeed'] );
+		$this->assertCount( 1, $result['wp-content']['post']['succeed'] );
+		$this->assertCount( 1, $result['wp-content']['page']['succeed'] );
+		$this->assertCount( 1, $result['wp-content']['tests']['succeed'] );
+		$this->assertCount( 6, $result['wp-content']['nav_menu_item']['succeed'] );
+		$this->assertFalse( isset( $result['wp-content']['sectests'] ) );
+
+		// Cleanup
+		unregister_post_type( 'tests' );
+		unregister_post_type( 'sectests' );
+	}
+
+	// Test the set default function. In this test we are also testing the import process by session ID.
+	public function test_set_default__import_by_session() {
+		// Arrange
+		$zip_path = __DIR__ . '/mock/sample-kit.zip';
+		$extraction_result = Plugin::$instance->uploads_manager->extract_and_validate_zip( $zip_path, [ 'json', 'xml' ] );
+		$session = $extraction_result['extraction_directory'];
+
+		// Act
+		$import = new Import( $session );
+
+		// Assert
+		$manifest = $import->get_manifest();
+		$expected_settings_include = [ 'templates', 'plugins', 'content', 'settings' ];
+		$expected_settings_referrer = 'local';
+		$expected_settings_selected_custom_post_types = [ 'tests', 'sectests' ];
+		$expected_settings_selected_override_conditions = [];
+		$expected_settings_selected_plugins = $manifest['plugins'];
+
+		$settings_include = $import->get_settings_include();
+		$settings_referrer = $import->get_settings_referrer();
+		$settings_selected_custom_post_types = $import->get_settings_selected_custom_post_types();
+		$settings_selected_override_conditions = $import->get_settings_selected_override_conditions();
+		$settings_selected_plugins = $import->get_settings_selected_plugins();
+
+		$this->assertEquals( $expected_settings_include, $settings_include );
+		$this->assertEquals( $expected_settings_referrer, $settings_referrer );
+		$this->assertEqualSets( $expected_settings_selected_custom_post_types, $settings_selected_custom_post_types );
+		$this->assertEquals( $expected_settings_selected_override_conditions, $settings_selected_override_conditions );
+		$this->assertEquals( $expected_settings_selected_plugins, $settings_selected_plugins );
+	}
+
+	public function test_constructor__import_by_not_existed_session() {
+		// Arrange
+		$elementor_tmp_directory = Plugin::$instance->uploads_manager->get_temp_dir();
+
+		// Expect
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage( 'missing-tmp-folder-error' );
+
+		// Act
+		$import = new Import( $elementor_tmp_directory . 'session-that-not-exits', [] );
+	}
+
+	// Test if the kit-library adaptor is running and performing correctly.
+	public function test_get_manifest__kit_library() {
```

Comment:
maybe just test that it imports properly?
Plus, what about envato?

Reply:
I don't want to upload full kit from kit library/Envato to our mocks.